web/src/echo.ts

```diff
@@ -431,6 +433,28 @@ export function edit_locally(message: Message, request: LocalEditRequest): Messa
     return message;
 }
 
+export function update_topic_hash_to_contain_with_term(message: Message): void {
+    // If the current filter consists only of channel and topic terms
+    // and the incoming message belongs to same narrow, we try to
+    // update the hash, to convert to permalink.
```

Comment:
This is useful when creating a new topic (when no message in topic, a normal topic link used in place of permalink) -- either you are narrowed to that new topic or you narrow to that upon insertion of that message.

Upon arrival of message, if current narrow is same as the stream and topic of the message, we add the `with` term to the current url